lib/Model/Importer/DebugInfo/PDBImporter.cpp

```diff
@@ -0,0 +1,1089 @@
+                << ReturnTypeIndex.getIndex());
+  }
+
+  return Error::success();
+}
+
+// LF_UNION (TPI)
+Error PDBImporterTypeVisitor::visitKnownRecord(CVType &Record,
+                                               UnionRecord &Union) {
+  TypeIndex FieldsTypeIndex = Union.getFieldList();
+  auto NewType = model::makeType<model::UnionType>();
+  NewType->OriginalName = Union.getName().str();
+  auto UnionTy = cast<model::UnionType>(NewType.get());
+
+  uint64_t Index = 0;
+  auto &TheFields = InProgressMemberTypes[FieldsTypeIndex];
+
+  // Handle an empty union, similar to 0-sized structs.
+  // Typedef it to void.
+  if (TheFields.size() == 0) {
+    auto TypedefTy = model::makeType<model::TypedefType>();
+    TypedefTy->OriginalName = Union.getName().str();
+
+    auto TypeDef = cast<model::TypedefType>(TypedefTy.get());
+    using Values = model::PrimitiveTypeKind::Values;
+    TypeDef
+      ->UnderlyingType = model::QualifiedType(Model
+                                                ->getPrimitiveType(Values::Void,
+                                                                   0),
+                                              {});
+
+    auto TyPath = Model->recordNewType(std::move(TypedefTy));
+
+    ProcessedTypes[CurrentTypeIndex] = TyPath;
+    return Error::success();
+  }
+
+  bool GeneratedOneFieldAtleast = false;
+  for (const auto &Field : TheFields) {
+    // Create new field.
+    uint64_t Offset = Field.getFieldOffset();
+    if (not ProcessedTypes.count(Field.getType())
+        and Field.getType().isSimple())
+      createPrimitiveType(Field.getType());
+
+    if (ProcessedTypes.count(Field.getType())) {
+      auto MaybeSize = ProcessedTypes[Field.getType()].get()->size();
+      uint64_t Size = 0;
+      if (not MaybeSize)
+        Size = 0;
+      else
+        Size = *MaybeSize;
+
+      if (Size == 0) {
+        // Skip 0-sized field.
+        continue;
+      }
+
+      GeneratedOneFieldAtleast = true;
+      auto &FieldType = UnionTy->Fields[Index];
+      FieldType.OriginalName = Field.getName().str();
+      model::QualifiedType Qty(ProcessedTypes[Field.getType()], {});
+      FieldType.Type = Qty;
+
+      Index++;
+    } else {
+      revng_log(DILogger,
+                "LF_UNION: Unknown field type " << Field.getType().getIndex());
+    }
+  }
+
+  if (GeneratedOneFieldAtleast) {
+    auto TyPath = Model->recordNewType(std::move(NewType));
+    ProcessedTypes[CurrentTypeIndex] = TyPath;
+  }
+
+  return Error::success();
+}
+
+Error PDBImporterTypeVisitor::visitKnownRecord(CVType &Record,
+                                               ArgListRecord &Args) {
+  InProgressArgumentsTypes[CurrentTypeIndex] = Args;
+  return Error::success();
+}
+
+// ==== Implementation of the Model type recordings ==== //
+
+// TODO: This can go into LLVM, but thre is an ongoing review that should
+// implement this.
+static uint64_t getSizeinBytes(TypeIndex TI) {
+  if (not TI.isSimple())
+    return 0;
```

Comment:
I suggest not to return `0` to represent an invalid state, it might get sum'd to some other legit value (due to some bug).
Consider returning `std::optional<uint64_t>` to be extra explicit.